meta-lmp-base/recipes-support/libp11/files/0001-from-list-Introduce-generic-keypair-generation-inter.patch

```diff
@@ -0,0 +1,804 @@
+From 140624f5062b24b4a1ac1097149c16003bc42993 Mon Sep 17 00:00:00 2001
+From: Ivan Stepic <Ivan.Stepic@bmw.de>
+Date: Thu, 8 Sep 2022 14:02:24 +0200
+Subject: [PATCH 1/4] [from list] Introduce generic keypair generation
+ interface and engine ctrl command
+
+As discussed in https://github.com/OpenSC/libp11/pull/379 and
+https://github.com/OpenSC/libp11/pull/378 we need a generic interface
+that supports multiple algorithms for key generation. Attempt was made
+to create a new keygen method and register it in PKCS11_pkey_meths() in
+p11_pkey.c (so that it's possible to generate keys using OpenSSL's
+EVP_PKEY_* API) but multiple design issues appeared. How and where do you
+pass the key ID, token label and alike was the first question. As
+suggested by the maintainer here:
+https://github.com/OpenSC/libp11/pull/379#issuecomment-820588833,
+app_data from EVP_PKEY_CTX was (mis)used and that worked well. The
+reason why this approach was abandoned is because a good (or bad) way
+to get a handle of the PKCS11_CTX_private, that is necessary for the
+Cryptoki call, was not found.
+The way other operations work is that they rely on the key being
+loaded *_first_* through ENGINE_load_public(private)_key because this
+is when the PKCS11_CTX gets initialized and a handle to
+PKCS11_OBJECT_private gets set to the ex_data of the underlying key.
+Key generation obviously cannot rely on that mechanism since key
+doesn't yet exist.
+
+Instead, a generic PKCS11_generate_key interface was made that
+takes a structure describing the key generation algorithm. For now
+it only contains simple options like curve name for ECC or number
+of bits for RSA key generation. This interface can then be used
+as any other PKCS11 wrapper interface or using the ENGINE control
+commands. Using it with ENGINE control commands is demonstrated in
+the new tests/keygen.c file.
+
+Code for ECC keygen was taken from:
+https://github.com/OpenSC/libp11/pull/379 and reworked to compile and
+work with some new additions to libp11 i.e. templates.
```

Comment:
```

Upstream-Status: Submitted [https://github.com/OpenSC/libp11/pull/474]
Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
```

Reply:
> Add some suggestions about the [Upstream-Status:](https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations:_Upstream-Status) that is something strictly related with the openembedded.

@ldts where do I add this info

Reply:
We add this info in the patch itself (in this case is same info for all the pacthes from the same PR) so when we do the next update it will help in droping/rebasing/update the patches.
There are some more examples in https://github.com/search?q=repo%3Afoundriesio%2Fmeta-lmp+Upstream-Status&type=code